Crashlytics/ProtoSupport/Protos/crashlytics.proto

```diff
@@ -47,206 +45,240 @@ message Session {
+          string name = 3;
 
-          // 128-bit object UUID matches Mach-O DWARF dSYM files */
-          optional bytes uuid = 4;
+          // 128-bit object UUID matches Mach-O DWARF dSYM files
+          bytes uuid = 4;
 
-          optional Architecture arch = 5;
+          Architecture arch = 5;
         }
         repeated BinaryImage binaries = 4;
       }  // message Application.Execution
 
       repeated CustomAttribute custom_attributes = 2;
 
-      optional bool background = 3; /* true if the app was in the background */
-      optional uint32 ui_orientation = 4;
+      // true if the app was in the background
+      bool background = 3;
+
+      uint32 ui_orientation = 4;
     }  // message Event.Application
 
-    optional Application app = 3;
+    Application app = 3;
 
     // Device physical properties recorded at event time
     message Device {
       // Battery level from 0.0 to 1.0
-      optional float battery_level = 1;
+      float battery_level = 1;
 
       // True when the proximity sensor is activated during the event
-      optional bool proximity_on = 3;
+      bool proximity_on = 3;
 
       // Device orientation during event
       // 1,2 = portrait
       // 3,4 = landscape
       // 5 = face up
       // 6 = face down
-      optional uint32 orientation = 4;
+      uint32 orientation = 4;
 
       // Total RAM used at event time, in bytes
-      optional uint64 ram_used = 5;
+      uint64 ram_used = 5;
 
       // Total disk used at event time, in bytes
-      optional uint64 disk_used = 6;
+      uint64 disk_used = 6;
     }
 
-    optional Device device = 5;
+    Device device = 5;
 
     // Developer-supplied log lines
     // Only those lines most recently logged prior to the event are captured
     message Log {
-      optional string content = 1;
+      string content = 1;
     }
 
-    optional Log log = 6;
+    Log log = 6;
   }  // message Event
   repeated Event events = 10;
 }
 
 // Map entry for developer-supplied custom attribute/value pairs
 message CustomAttribute {
 
-  required string key = 1;
+  string key = 1;
 
-  required string value = 2;
+  string value = 2;
 }
+
+message Report {
+  // SDK version that generated this session
+  string sdk_version = 1;
+
+  // GMP App ID
+  string gmp_app_id = 3;
+
+  // General device type which generated the Event
+  google_crashlytics.Session.Platform platform = 4;
+
+  // Unique device generated guid for application install. Equivalent to
+  // Session.app.installation_uuid
+  string installation_uuid = 5;
+
+  // App build version. Equivalent to Session.app.version.
+  string build_version = 6;
+
+  // Developer-supplied application version. Equivalent to
+  // Session.app.display_version.
+  string display_version = 7;
+
+  // We store the actual crash information in a separate field for each client
+  // type. This is a temporary measure until we've come up with a generic
+  // cross-platform format.
+  message payload {
```

Comment:
should the message type begin with a capital letter?

Reply:
I am going to leave this as is since I want this proto to be consist with the backend proto (as much as possible)